algorithms/closest_pair.py

```diff
@@ -84,7 +84,12 @@ def _distance_between(point_1, point_2):
 
```

Comment:
re: `_calculate_closest_split_pair`, the `y_sorted` is used to iterate through to build the list of pairs that have x values within the delta range. This was going from what the video said to use, although thinking about it I don't know why you can't just use the `x_sorted` list...

Not sure where you mean when you say 'filtering one list based on a completely different one.'

I also have a thought that  `_caluculate_closest_split_pair` should be run for each recursion. Currently it only gets run once for the whole thing. Not sure how correct that it...